clap_complete/src/dynamic/completer.rs

```diff
@@ -71,8 +72,37 @@ pub fn complete(
         }
 
         if is_escaped {
-            pos_index += 1;
-            state = ParseState::Pos(pos_index);
+            match state {
+                ParseState::ValueDone => {
+                    // Begin parsing a new positional argument after `--`.
+                    let num_args = parse_positional(current_cmd, pos_index);
+                    match num_args {
+                        Some(num_args) => {
+                            state = ParseState::Pos(1);
+                            if num_args <= 1 {
+                                pos_index += 1;
+                            }
+                        }
+                        None => {
+                            state = ParseState::ValueDone;
```

Comment:
Won't `ValueDone` put is back into parsing options and subcommands on the next iteration?